src/sqlfluff/dialects/dialect_redshift.py

```diff
@@ -56,3 +65,214 @@ class FunctionSegment(BaseSegment):
 
     type = "function"
     match_grammar = ansi_dialect.get_segment("FunctionSegment").match_grammar.copy()
+
+
+@redshift_dialect.segment()
+class ColumnEncodingSegment(BaseSegment):
+    """ColumnEncoding segment.
+
+    Indicates column compression encoding.
+
+    As specified by: https://docs.aws.amazon.com/redshift/latest/dg/c_Compression_encodings.html
+    """
+
+    type = "column_encoding_segment"
+
+    match_grammar = OneOf(
+        "RAW",
+        "AZ64",
+        "BYTEDICT",
+        "DELTA",
+        "DELTA32K",
+        "LZO",
+        "MOSTLY8",
+        "MOSTLY16",
+        "MOSTLY32",
+        "RUNLENGTH",
+        "TEXT255",
+        "TEXT32K",
+        "ZSTD",
+    )
+
+
+@redshift_dialect.segment()
+class ColumnAttributeSegment(BaseSegment):
+    """Redshift specific column attributes.
+
+    As specified in https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_TABLE_NEW.html
+    """
+
+    type = "column_attribute_segment"
+
+    match_grammar = AnyNumberOf(
+        Sequence("DEFAULT", Ref("ExpressionSegment")),
+        Sequence(
+            "IDENTITY",
+            Bracketed(
+                Delimited(Ref("NumericLiteralSegment"), Ref("NumericLiteralSegment"))
+            ),
+        ),
+        Sequence(
+            "GENERATED",
+            "BY",
+            "DEFAULT",
+            "AS",
+            "IDENTITY",
+            Bracketed(
+                Delimited(Ref("NumericLiteralSegment"), Ref("NumericLiteralSegment"))
+            ),
+        ),
+        Sequence("ENCODE", Ref("ColumnEncodingSegment")),
+        "DISTKEY",
+        "SORTKEY",
+        Sequence("COLLATE", OneOf("CASE_SENSITIVE", "CASE_INSENSITIVE")),
+    )
+
+
+@redshift_dialect.segment(replace=True)
+class ColumnConstraintSegment(BaseSegment):
+    """Redshift specific column constraints.
+
+    As specified in https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_TABLE_NEW.html
+    """
+
+    type = "column_constraint_segment"
+
+    match_grammar = AnyNumberOf(
+        OneOf(Sequence("NOT", "NULL"), "NULL"),
+        OneOf("UNIQUE", Sequence("PRIMARY", "KEY")),
+        Sequence(
+            "REFERENCES",
+            Ref("TableReferenceSegment"),
+            Sequence(Bracketed(Ref("ColumnReferenceSegment"))),
```

Comment:
I'm not sure this line is quite right. I think you don't need the Sequence(), and the Bracketed needs an optional=True